sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/WithTimestamps.java

```diff
@@ -101,7 +101,7 @@ public Duration getAllowedTimestampSkew() {
   public PCollection<T> apply(PCollection<T> input) {
     return input
         .apply("AddTimestamps", ParDo.of(new AddTimestampsDoFn<T>(fn, allowedTimestampSkew)))
-        .setTypeDescriptorInternal(input.getTypeDescriptor());
+        .setCoder(input.getCoder());
```

Comment:
I think this still improves the readability - this is explicit, and insulated against any changes to coder  inference (and gets away from using `Internal` methods)